src/main/main.rs

```diff
@@ -27,7 +29,7 @@ pub extern fn main() {
 
 
     let processes = unsafe {
-        process::process::load_processes(&_sapps)
+        process::process::load_processes(&_sapps, &_eapps)
```

Comment:
Ah, this is actually incorrect (although there is no reasonable way to have known this from the code). &_eapps is only meaningful if the user-level processes are compiled into the kernel (which is true on some platforms but not generally, and definitely not on the Firestorm). Otherwise, processes are programmed onto flash out of band (on the Firestorm, via `tools/program/firestorm.py`) and the `.apps.*` sections have size 0 and `_eapps == _sapps`.

In this case, we know we've reached the last app to load because the length field is 0 (in practice we do this by just padding the last app with `0x000000`).